lib/elixir_analyzer.ex

```diff
@@ -163,25 +175,24 @@ defmodule ElixirAnalyzer do
   defp do_init(%{file: nil} = params, exercise_config) do
     meta_config = Path.join(params.path, @meta_config) |> File.read!() |> Jason.decode!()
     relative_code_path = meta_config["files"]["solution"] |> hd()
-    full_code_path = Path.join(params.path, relative_code_path)
-
-    code_path = Path.dirname(full_code_path)
-    code_file = Path.basename(full_code_path)
+    code_path = Path.join(params.path, relative_code_path)
 
-    exemplar_path =
-      case meta_config["files"]["exemplar"] do
-        [path | _] -> Path.join(params.path, path)
+    {exercice_type, exemploid_path} =
+      case meta_config["files"] do
+        %{"exemplar" => [path | _]} -> {:concept, Path.join(params.path, path)}
+        %{"example" => [path | _]} -> {:practice, Path.join(params.path, path)}
```

Comment:
@neenjaw do I remember correctly that your initial assumption was to just ignore multi file submissions? I think that was totally reasonable as they are very rare. It would be cool to at least get the basic common checks working for multi file solutions (e.g. proper name casing, indentation) but I wouldn't spend too much effort on making it perfect.

Reply:
When I wrote that line of code, example files were only a single file. If that's not the case anymore, then makes sense to change strategy.

Reply:
Example files are still a single file, but the checks only look at the file mentioned in `meta_config["files"]["solution"]`. In case students submit several files, they are not looked at. And might actually get a compiler error as an analyzer comment, now that I think about it. Could you re-submit one of your multiple file solutions and check?

Reply:
Compiled elixir is still dynamically linked at run time so it just needs to be legal syntax, not runnable code,  for code to compile successfully.

```
iex(1)> Code.compile_string("""
...(1)> defmodule MyModule do
...(1)>   def foo do
...(1)>     SomeNonExistentModule.bar(2)
...(1)>   end
...(1)> end
...(1)> """)
[
  {MyModule,
   <<70, 79, 82, 49, 0, 0, 4, 236, 66, 69, 65, 77, 65, 116, 85, 56, 0, 0, 0,
     171, 0, 0, 0, 16, 15, 69, 108, 105, 120, 105, 114, 46, 77, 121, 77, 111,
     100, 117, 108, 101, 8, 95, 95, 105, 110, 102, 111, ...>>}
]
```

Reply:
That's true enough, although not foolproof.

```
iex(42)> Code.compile_string("""
...(42)> defmodule MyModule do
...(42)> import SomeNonExistentModule    
...(42)> def foo do
...(42)>   bar(2)
...(42)> end
...(42)> end
...(42)> """ )
** (CompileError) nofile:2: module SomeNonExistentModule is not loaded and could not be found
```